### PR TITLE
IAM - Generate CreateAccessKey Action

### PR DESCRIPTION
**Pull request by amenophis**

### CODE REVIEW

File: src/Service/Iam/src/Enum/statusType.php [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace AsyncAws\Iam\Enum;
+
+final class statusType
```
Comment (Nyholm) [inline src/Service/Iam/src/Enum/statusType.php:5]: This is strange. The AWS specification names this `statusType` and not `StatusType`. 

We can: 
1. Force first character in a class to be capitalised. 
2. Ignore it
3. Add a patch to the manifest.

Comment (amenophis) [reply src/Service/Iam/src/Enum/statusType.php:5]: Will fix this globally in another PR

**Review by Nyholm — APPROVED** [review]
Im happy with this. 

Thank you

File: src/Service/Iam/src/Enum/statusType.php (hunk 1 above)
Comment (Nyholm) [reply src/Service/Iam/src/Enum/statusType.php:5]: FYI @jderusse

Comment (Nyholm) [reply src/Service/Iam/src/Enum/statusType.php:5]: This might be a fix: #815

File: manifest.json [hunk 2]
```diff
@@ -120,7 +120,7 @@
             "example": "https://raw.githubusercontent.com/aws/aws-sdk-php/${LATEST}/src/data/iam/2010-05-08/examples-1.json",
             "api-reference": "https://docs.aws.amazon.com/IAM/latest/APIReference",
             "methods": [
-                "AddUserToGroup",
```
Comment (jderusse) [inline manifest.json:123]: Why this change? did you removed the method?

File: src/Service/Iam/tests/Unit/Result/CreateAccessKeyResponseTest.php [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+          </ResponseMetadata>
+        </CreateAccessKeyResponse>');
+
+        $client = new MockHttpClient($response);
+        $result = new CreateAccessKeyResponse(new Response($client->request('POST', 'http://localhost'), $client, new NullLogger()));
+
+        self::assertSame('AKIAIOSFODNN7EXAMPLE', $result->getAccessKey()->getAccessKeyId());
+        self::assertSame('2020-10-13 17:38:43.000000', $result->getAccessKey()->getCreateDate()->format('Y-m-d H:i:s.u'));
+        self::assertSame('wJalrXUtnFEMI/K7MDENG/bPxRfiCYzEXAMPLEKEY', $result->getAccessKey()->getSecretAccessKey());
+        self::assertSame('Active', $result->getAccessKey()->getStatus());
```
Comment (jderusse) [inline src/Service/Iam/tests/Unit/Result/CreateAccessKeyResponseTest.php:38]: Please use constant when available. It prove the const works.

```suggestion
        self::assertSame(statusType::ACTIVE', $result->getAccessKey()->getStatus());
```

**Review by jderusse — COMMENTED** [review]
Thanks for this contribution. I've just a small comment.
Everything else is fine when #815 would be merged and applied to this PR

File: src/Service/Iam/tests/Unit/Result/CreateAccessKeyResponseTest.php (hunk 3 above)
Comment (amenophis) [reply src/Service/Iam/tests/Unit/Result/CreateAccessKeyResponseTest.php]: Done

File: manifest.json (hunk 2 above)
Comment (amenophis) [reply manifest.json:123]: mistake ... Fixed !

**Comment by amenophis** [comment]
All good, waiting for #815 to fix the status casing

File: src/Service/Iam/src/Enum/statusType.php (hunk 1 above)
Comment (jderusse) [reply src/Service/Iam/src/Enum/statusType.php:5]: #815 merge. Could you please regenerate the method to apply the change @amenophis?

**Comment by amenophis** [comment]
@jderusse @Nyholm It's green !

**Review by Nyholm — APPROVED** [review]
Cool cool cool .

Thank you
